packages/leancode_lint/lib/helpers.dart

```diff
@@ -217,3 +219,72 @@ extension LintRuleNodeRegistryExtensions on LintRuleNodeRegistry {
     });
   }
 }
+
+bool isExpressionExactlyType(
+  Expression expression,
+  String typeName,
+  String packageName,
+) {
+  if (expression.staticType case final type?) {
+    return TypeChecker.fromName(
+      typeName,
+      packageName: packageName,
+    ).isExactlyType(type);
+  }
+  return false;
+}
+
+bool isInstanceCreationExpressionOnlyUsingParameter(
+  InstanceCreationExpression node, {
+  required String parameter,
+  List<String> ignoredParameters = const [],
+}) {
+  var hasParameter = false;
+
+  for (final argument in node.argumentList.arguments) {
+    if (argument
+        case NamedExpression(
+              name: Label(label: SimpleIdentifier(name: final argumentName)),
+            ) &&
+            final argument) {
+      if (ignoredParameters.contains(argumentName)) {
+        continue;
+      } else if (argumentName == parameter &&
+          argument.staticType?.nullabilitySuffix !=
+              NullabilitySuffix.question) {
+        hasParameter = true;
+      } else {
+        // Other named arguments are not allowed
+        return false;
+      }
+    } else {
+      // Other arguments are not allowed
+      return false;
+    }
+  }
+  return hasParameter;
+}
+
+class ChangeWidgetNameFix extends DartFix {
+  ChangeWidgetNameFix(this.widgetName);
+
+  final String widgetName;
+
+  @override
+  void run(
+    CustomLintResolver resolver,
+    ChangeReporter reporter,
+    CustomLintContext context,
+    error.AnalysisError analysisError,
+    List<error.AnalysisError> errors,
+  ) {
+    reporter
+        .createChangeBuilder(message: 'Replace with $widgetName', priority: 1)
+        .addDartFileEdit(
+          (builder) => builder.addSimpleReplacement(
+            analysisError.sourceRange,
```

Comment:
This assumes the lint will report an error at the constructor's name range. Since this is in the generic `helpers.dart` file this assumption should be documented

Reply:
I've added docs for this helper. I've mentioned what's needed there to use this class.